test/AutoRest.Shared.Tests/RequestUriBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class RequestUriBuilderExtensionsTests
```

Comment:
It's good to see a dedicated test for this class -- thanks for adding it!  Would it make sense to add tests for the new cases to the test `PathsClient` and `PathsRestClient` as well?

Reply:
Unfortunately we must remove this test case because of this issue: https://github.com/Azure/azure-sdk-for-net/issues/35828
The escaping logic is running differently in net6.0 and net461 therefore some of the test cases here will never pass.
Because of the same reason, I have to say that we may not add a case for `PathsClient` or `PathsRestClient` because they might have the same issue.
Therefore to minimize the change, we will add a new test case for the `ConvertToString` method because
- The `RequestUriBuilder` in `Azure.Core` only accepts strings or `Span<char>` for `AppendPath` and `AppendQuery`
therefore the responsibility of the code in `Generator.Shared` is just making sure we have the correct conversion from various types to string, and then `RequestUriBuilder` in Azure.Core should handle them correctly.